chia/cmds/data.py

```diff
@@ -146,12 +146,14 @@ def get_value(
 @create_rpc_port_option()
 @create_fee_option()
 @options.create_fingerprint()
+@click.option("-p", "--publish_on_chain", help="Publish the result on chain", type=bool, default=True)
```

Comment:
I would suggest not using a short option and using "--publish/--no-publish" as a flag